src/main/java/org/testng/annotations/Parameters.java

```diff
@@ -14,7 +13,7 @@
  * @author <a href="mailto:cedric&#64;beust.com">Cedric Beust</a>
  */
 @Retention(RUNTIME)
-@Target({METHOD, CONSTRUCTOR, TYPE })
+@Target({METHOD, TYPE})
 public @interface Parameters {
```

Comment:
@merapy - I am only talking about breaking an existing functionality that is already available to users. Injecting parameters into a constructor via the `@Parameters` annotation is pretty much the same kind as what people currently do when they use it with  a `@Test` annotation [ Scenarios wherein a test class depends on a parameter for all its test method working.. for.e.g., to figure out what data source to be used by a data provider that powers the test methods within the test class ].